### DISCUSSION TITLE
Allow (custom) nodes a way to patch/unpatch the model

### DISCUSSION THREAD

**Discussion by BlenderNeko** [Comfy-Org/ComfyUI#408] [Ideas]
I'm trying to figure out how to best implement [DAAM](https://github.com/castorini/daam) in a couple of custom nodes. DAAM would require changing the cross attention forward passes through something like monkey patching. I could do this patching/unpatching in a custom sampler node, emulating how this is now happening behind the scenes with control nets? But it would maybe be nicer if there was some kind of structure in place for other nodes to append their own patch/unpatch functions to that get run before/after sampling. thoughts? advice? etc.

**Comment by comfyanonymous** [comment]
Yes I'm planning on adding a modular way to add things to the attention stuff because I'm going to need to do that for hypernetworks, GLIGEN if I decide to implement it and a bunch of other stuff.

It's not going to involve monkey patching because I hate hooks.
